### PR TITLE
Validate Frequency transition on Baremetal.

### PR DESCRIPTION
**Pull request by shriyak**
The script validate frequency transition for userspace governor. Set the governor to userspace,
select the random frequency for random cpu and verify it using ppc64_cpu --frequency tool.
The difference is between frequency setting using cpupower tool and read using ppc64_cpu tool
is calculated and asserted if it crosses threshold value.
Test is applicable on baremetal platform.

Signed-off-by: Shriya <shriyak@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by shriyak** [comment]
Test_results.
[em_cpufreq_logs.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/1367536/em_cpufreq_logs.txt)

File: cpu/em_cpufreq.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            process.run("cpupower frequency-set -f %s"
+                        % self.get_random_freq())
+            freq_set = self.cpu_freq_path("cpuinfo_cur_freq")
+            freq_read = process.system_output("ppc64_cpu --frequency -t 5"
+                                              "| grep 'avg:' | awk "
+                                              "'{print $2}'", shell=True)
+            freq_read = float(freq_read) * (10 ** 6)
+            diff = float(freq_read) - float(freq_set)
+            self.log.info(" Difference is %s" % diff)
+            if (diff > threshold):
```
Comment (harish-24) [inline cpu/em_cpufreq.py]: Braces are not necessary here. Better to remove.

File: cpu/em_cpufreq.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                self.log.info("Frequency set and frequency read differs :"
+                              "%s %s " % (freq_set, freq_read))
+                self.error_count += 1
```
Comment (harish-24) [inline cpu/em_cpufreq.py]: `self.error_count` has not been used elsewhere.

File: cpu/em_cpufreq.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+                self.log.info("Works as expected for iteration %s " % var)
+
+    def get_random_freq(self):
+        """
+        Get random frequency from list
+        """
+        cmd = "scaling_available_frequencies"
+        return random.choice(self.cpu_freq_path(cmd).split(' '))
+
+    def get_random_cpu(self):
```
Comment (harish-24) [inline cpu/em_cpufreq.py]: Method could be static without the use of `self`

File: cpu/em_cpufreq.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        """
+        Set the governor to userspace
+        choose random frequency and cpu
+        validate the frequency is set using ppc64_cpu tool.
+        """
+        self.cpu = 0
+        threshold = 10000
+        output = process.run("cpupower frequency-set -g userspace", shell=True)
+        cur_governor = self.cpu_freq_path('scaling_governor')
+        if (output.exit_status == 0) and ('userspace' == cur_governor):
```
Comment (harish-24) [inline cpu/em_cpufreq.py]: braces are not required here also. Better to place constant `'userspace'` in right hand side.

File: cpu/em_cpufreq.py [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+import os
+import random
+import platform
+from avocado import Test
+from avocado import main
+from avocado.utils import process, distro, cpu
+from avocado.utils.software_manager import SoftwareManager
+
+
+class cpufreq(Test):
```
Comment (harish-24) [inline cpu/em_cpufreq.py]: `Cpufreq` would be a better name by python standards for class.

**Comment by shriyak** [comment]
@harish-24 : Sure let me update the script. Thanks

**Comment by shriyak** [comment]
closing this PR and raising new PR

**Pull request closed without merging**